common/test/acceptance/pages/lms/teams.py

```diff
@@ -502,6 +502,10 @@ def team_leave_link_present(self):
 
     def click_leave_team_link(self, remaining_members=0, cancel=False):
         """ Click on Leave Team link"""
+        self.wait_for(
+            lambda: self.team_leave_link_present,
+            description="Leave Team button did not become present"
+        )
         self.q(css='.leave-team-link').first.click()
```

Comment:
Note that there is a helper click_css method in utils. I'd recommend you use that, as it does waiting for you (and possibly other things).